doc/adr/0004-nats-headers.md

```diff
@@ -55,26 +55,28 @@ The server that is able to send and receive headers will specify so in it's [`IN
 Messages that include a header have a `HPUB` protocol:
 
 ```
-HPUB SUBJECT REPLY 23 30␍␊NATS/1.0␍␊Header: X␍␊␍␊PAYLOAD␍␊
-HPUB SUBJECT REPLY 23 23␍␊NATS/1.0␍␊Header: X␍␊␍␊␍␊
+HPUB SUBJECT REPLY 13 20␍␊NATS/1.0␍␊Header: X␍␊␍␊PAYLOAD␍␊
```

Comment:
Why did `23 30` change to `13 20`?

I ran this code
```go
nc.PublishMsg(&nats.Msg{
	Subject: "SUBJECT",
	Header: map[string][]string{
		"Header": []string{"X"},
	},
	Data: []byte("PAYLOAD"),
	Reply: "REPLY",
})
```
but the logs showed this.
```
[HPUB SUBJECT REPLY 23 30]
MSG_PAYLOAD: ["NATS/1.0\r\nHeader: X\r\n\r\nPAYLOAD"]
```

Logs are from latest nats-server.

Reply:
So I likely misunderstood the notes. The doc says this:
> HDR_LEN includes the entire serialized header

So 13 is the length of `Header: X␍␊␍␊` It was not clear from the doc that the ten bytes of `NATS/1.0␍␊` was to be included as part of the header length. I will change it back and add clarification.